### DISCUSSION TITLE
What does a protocol mean?

### DISCUSSION THREAD

**Discussion by joewheaton** [Riverscapes/QRiS#269] [Ideas & Feature Requests]
@philipbaileynar asked:
1. What does a protocol mean?
2. What is anatomy of a protocol? Protocol, Methods, Layers, Attributes, Metrics
3. How does it change over time?
4. How does a user create a protocol from scratch?
5. How does a user create a protocol from an old protocol?
6. How does a user add or update methods within a protocol?
7. Who can author a protocol?
8. Are all protocols visible, editable, consumable by anyone?

**Comment by philipbaileynar** [comment]
# Concept 1

I have this vision that a protocol is merely a lens through which you view your data. In other words, you can "apply" any protocol at any time to a QRiS project. It doesn't change the database structure, it merely prompts you to capture slightly different data. 

You can start a QRiS project with protocol A and then download and apply protocol B and then revert to protocol A again. These two protocols might require different layers or attributes. QRiS is going to prompt you to always capture the data for the "active" protocol, but allow any other data for previous ("dormant") protocols. These layers and fields are still part of the project, they are just ignored.

# Concept 2

Perhaps a protocol is applied to a DCE as the DCE is created. It can be changed, but the key point is that the protocol is associated with the DCE and not the QRiS package overall. This allows for a QRiS project that spans many field seasons to have DCEs from the first year associated with one protocol and then subsequent years use an updated version of the protocol while leaving the original DCEs untouched.
